src/core/qgsdistancearea.cpp

```diff
@@ -58,6 +58,8 @@ bool QgsDistanceArea::willUseEllipsoid() const
 void QgsDistanceArea::setSourceCrs( const QgsCoordinateReferenceSystem &srcCRS )
 {
   mCoordTransform.setSourceCrs( srcCRS );
+  setEllipsoid( sourceCrs().ellipsoidAcronym() );
```

Comment:
remove this - sourceCrs is TOTALLY 100% independent of the ellipsoid choice.

The source CRS is just used as an aid to ensure that points passed to QgsDistanceArea are correctly transformed from WHATEVER their CRS is to the CRS which matches the ellipsoid. (E.g. It's "source" CRS - not "dest" CRS).